lib/barzer/barzer_objects.py

```diff
@@ -0,0 +1,222 @@
+        self.relevance = data.get('rel', 0)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __str__(self):
+        return 'entity:{}.{}.{}'.format(self.scope, self.category, self.id)
+
+    def ent_id(self):
+        return EntityId(self.eclass, self.subclass, self.id)
+
+class Range(Bead):
+    type_name = "range"
+    def __init__(self, data):
+        super(Range, self).__init__(data)
+        self.rangetype = data.get('rangetype', 'real')
+        self.order = data.get('order')
+        self.lo = data.get('lo')
+        self.hi = data.get('hi')
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __str__(self):
+        return '({}, {})'.format(str(self.lo), str(self.hi))
+
+    def get_as_type_pair(self, the_type=float):
+        try:
+            lo = the_type(self.lo)
+        except Exception:
+            lo = None
+
+        try:
+            hi = the_type(self.lo)
+        except Exception:
+            hi = None
+
+        if lo is not None and hi is not None:
+            return (lo, hi) if lo < hi else (hi, lo)
+        else:
+            return (lo, hi)
+
+    def get_as_single_number(self, num_type=float):
+        pair = self.get_as_type_pair(the_type=num_type)
+        if pair[0] is None or pair[1] is None:
+            return None
+        else:
+            return (pair[0] + pair[1])/2
+
+class EVR(EntityBase):
+    """ entity value range """
+    type_name = "evr"
+    def __init__(self, data):
+        super(EVR, self).__init__(data)
+        self.ent = Entity(data['ent'])
+        self.values = [BeadFactory.make_bead_from_dict(x) for x in data.get('values', [])]
+
+    @property
+    def eclass(self):
+        return self.ent
+
+    @property
+    def subclass(self):
+        return self.subclass
+
+    @property
+    def id(self):
+        return self.id
+
+    def __str__(self):
+        return '{}:({} [{}])'.format(
+            self.type_name, str(self.ent), ','.join(str(x) for x in self.values))
+
+    def ent_id(self):
+        return self.ent.id
+
+    def iterate_type(self, the_type=Number):
+        """ yields all values of type the_type.
+        NOTE: the_type can also be a tuple
+        """
+        for v in self.values:
+            if isinstance(v, the_type):
+                yield v
+
+class ERC(EntityBase):
+    type_name = "erc"
+    """ entity range combo """
+    def __init__(self, data):
+        super(ERC, self).__init__(data)
+        self.ent = Entity(data['ent'])
+        self.range = Range(data.get('range'))
+
+    def __str__(self):
+        return '{}:({}{})'.format(self.type_name, str(self.ent), self.range)
+
+    def ent_id(self):
+        return self.ent.ent_id()
+
+class Time(ValueBead):
+    type_name = "time"
+    """ time of day """
+
+
+class Timestamp(Bead):
+    type_name = "timestamp"
+    """ """
+    def __init__(self, data):
+        super(Timestamp, self).__init__(data)
+        self.date, self.time = data['date'], data['time']
```

Comment:
valid Timestamp must have both `date` and `time` so no, not `get`
as to converting to `datetime` - that's a separate option but I want to preserve the fact that it's the bead of type `Date`, `Time` or `DateTime`